signer/src/request_decider.rs

```diff
@@ -329,20 +311,6 @@ where
     }
 
     async fn can_accept_deposit_request(&self, req: &model::DepositRequest) -> Result<bool, Error> {
-        // If the deposit request is larger than the allowed per-deposit cap
-        // then no signer should accept this deposit.
-        if let Some(per_deposit_cap) = self.context.state().get_current_limits().per_deposit_cap() {
-            if req.amount > per_deposit_cap.to_sat() {
-                tracing::info!(
-                    txid = %req.txid,
-                    outpoint = %req.outpoint(),
-                    amount = %req.amount,
-                    "deposit request exceeds per-deposit cap"
-                );
-                return Ok(false);
-            }
-        }
-
```

Comment:
I'm having second thoughts about removing this check. I believe the per-withdrawal and per-deposit cap validation should remain here, and I'll reintroduce the code.
I'll update the comments to clarify that can_accept means the request was both approved by the blocklist and within the cap limits.

Reply:
Yeah I think it's good to have the checks here as well, because if enough signers vote `can_accept = false` then the coordinator won't even try to package them.

Note that I think we need to check `amount + max_fee` here, otherwise the user could potentially specify `10 000` with max fee of `10 000 000` and circumvent the check?

I might consider re-adding the limits state in context and updating the limits from Emily once in the block observer upon new Bitcoin block though, because now the limit-pulling from Emily will be scattered a bit all over the place. If it's updated when receiving a new Bitcoin block then the limits will apply to all validations done by the signer within that block (request-decider, tx-signer and tx-coordinator).

Reply:
mmm something to consider is that after https://github.com/stacks-network/sbtc/pull/968 we are no longer voting on requests we already voted for, so if we want to reconsider deposits after we change the limit we should *not* vote against them in here.

Reply:
My only concern with not voting against them is that I guess the coordinators will perpetually try to package them until the total supply decreases (which can't currently happen since we don't have withdrawals) and they manage to get squeezed in, so the list of pending deposits could potentially grow pretty big.

Reply:
As discussed at the standup, we are going to remove this check from here, but we will introduce a max lock filter (in a separate PR) and a per-deposit-cap filter in the coordinator (implemented in https://github.com/stacks-network/sbtc/pull/967/commits/fe55499a492a0b1b5f94ac40f2c1438f50478ef3 ).